client/components/field.js

```diff
@@ -31,7 +27,7 @@ class Field extends Component {
     return this.props.onSave && this.props.onSave(value);
   }
 
-  mapOptions(options = []) {
+  mapOptions = (options = []) => {
```

Comment:
Any reason this one function has changed syntax from the rest of the file?

Reply:
I believe it needs to be bound to `this`, I didn't actually check. We were binding to `this` in a constructor previously, so I deleted that

Reply:
disregard previous comment - it needs to be bound to `this`

Reply:
Surely it's already bound to `this` on the basis that it's a class method?

Or is it because we do `someprop={this.mapOptions}` somewhere else (and thus unbinding)?

Reply:
I'll check :)

Reply:
It makes sense (hence my merging)

My preference is generally to arrow-function-ify at prop declare time i.e. `someProp={() => this.mapOptions()}`

Reply:
If you do that you are redefining the function every time the child is rendered - `() => {} !== () => {}` so it will appear the prop has changed and trigger a re-render. We should use handle functions when passing a function as a prop:

```js
handleDoSomething = val => {
  this.setState({ val });
}

<Component
  doSomething={handleDoSomething}
/>
```